tasty/skyspark/point_mapper.py

```diff
@@ -0,0 +1,148 @@
+        return self.root
+
+    def generate_point_tree(self, filename: str):
+        """
+        Generates/populates a PointTree based on a schema definition file (e.g. 'schemas/haystack/defs_3_9_10.ttl') which
+        describes the desired ontology.
+
+        :param filename: the filepath/filename from which to generate the PointTree
+        """
+        # parse the file to an rdf graph
+        graph = tg.get_versioned_graph(tc.HAYSTACK, tc.V3_9_10)
+        graph.parse(filename, format='turtle')
+
+        # get root of graph from the rdf graph
+        root_uri = tc.PHIOT_3_9_10[self.root_type]
+        count = 0
+        for s, p, o in graph.triples((root_uri, RDF.type, OWL.Class)):
+            count += 1
+            # create new root node
+            # for now passing in a type of "point" - the load_tags method will ignore this so no tags are added to the root
+            self.root = PointNode("point", None)
+
+        # assert that there can only be one root value
+        assert count == 1
+
+        # now iterate over all items which are a subclass of "point" and recursively add all subnodes
+        self.add_subnodes(graph, self.root, root_uri)
+
+    def add_subnodes(self, graph, parentNode: 'PointNode', parentUri):
+        """
+        Adds all subnodes of a given parent node to the PointTree. Given an rdf graph and a parent node in that graph,
+        this method adds subtypes of the root node as children to the root, and recursively does the same for each child
+        node untill all subtypes are discovered and the PointTree is fully populated.
+
+        :param graph: the rdf graph describing the given ontology from which to construct the PointTree
+        :param parentNode: the parent node as a PointNode, for which all subtypes should be added
+        :param parentUri: the uri of the parent node in the rdf graph
+        """
+        for s, p, o in graph.triples((None, RDFS.subClassOf, parentUri)):
+            # print(s[s.find('#')+1:])
+            new_node = PointNode(s[s.find('#') + 1:], parentNode)
+            parentNode.add_child(new_node)
+            self.add_subnodes(graph, new_node, s)
+
+    def determine_first_class_point_type(self, root: 'PointNode', input_tags) -> 'PointNode':
+        """
+        Determines the first-class-point-type based on a given set of tags starting at the given root node.
+        This method looks at the root node and then recursively at each subnode untill the set of tags stops
+        matching the point type.
+
+        :param root: the node at which to begin checking against
+        :param input_tags: a list of the tags to use to identify the point type
+        """
+        # iterate over children node
+        for node in root.children:
+
+            counter = 0
+            # iterate over the node tags
+            for tag in node.tags:
+                # if the tag is not in the input tag set, break and go to next node
+                if(tag not in input_tags):
```

Comment:
I think to avoid using `break` in the middle of this you could reverse the logic and have:
```
if tag in input_tags:
    #Do something
```
If the condition is not met python will just continue by default.  I think - def check me on that though.

Reply:
Yea you're right that's cleaner. I just have to move the following logic into that if-statement.